float32 origin_x
float32 origin_y
float32 height
float32 width